c/misra/test/rules/RULE-17-11/test.c

```diff
@@ -0,0 +1,91 @@
+#include "stdlib.h"
+
+_Noreturn void test_noreturn_f1(int i) { // COMPLIANT
+  abort();
+}
+
+void test_noreturn_f2(int i) { // NON_COMPLIANT
+  abort();
+}
+
+_Noreturn void test_noreturn_f3(int i) { // COMPLIANT
+  if (i > 0) {
+    abort();
+  }
+  exit(1);
+}
+
+void test_noreturn_f4(int i) { // NON_COMPLIANT
+  if (i > 0) {
+    abort();
+  }
+  exit(1);
+}
+
+void test_noreturn_f5(int i) { // COMPLIANT
+  if (i > 0) {
+    return;
+  }
+  exit(1);
+}
+
+void test_noreturn_f6(int i) { // COMPLIANT
+  if (i > 0) {
+    abort();
+  }
+  if (i < 0) {
+    abort();
+  }
+}
+
+void test_noreturn_f7(int i) { // COMPLIANT
+  if (i > 0) {
+    abort();
+  }
+}
+
+void test_noreturn_f8(int i) { // NON_COMPLIANT
```

Comment:
Can we also add a test case here for a `_Noreturn` function calling another `_Noreturn` function declared by us? I don't know if the control flow liveness reachability takes `_Noreturn` into account or not, so it would be good to validate.

Reply:
Same for 17.9.